### DISCUSSION TITLE
How to take joystick commands from FlightGear as inputs into JSBSim to control the model?

### DISCUSSION THREAD

**Discussion by adeelahsaanawan** [JSBSim-Team/jsbsim#899] [Q&A]
I have interfaced JSBSim with FlightGear. I want to control the aircraft using joystick? How can I do that.

Kindly guide!

**Comment by seanmcleod** [comment]
You haven't specified how exactly you're interfacing JSBSim with FlightGear.

Typically FlightGear loads JSBSim itself and FlightGear has code to read joystick inputs and FlightGear will then pass those inputs into JSBSim.

**Comment by adeelahsaanawan** [comment]
Through UDP protocol

**Comment by adeelahsaanawan** [comment]
Can you please send me the link to that code in which I can use flight gear. The thing is I am running JSBSim standalone and and want my joystick to communicate with JSBSim model, but you can advise me on that.

**Comment by seanmcleod** [comment]
Do you have a specific reason to run JSBSim externally to FlightGear?

> Can you please send me the link to that code in which I can use flight gear.

I'm not familiar with the FlightGear code, you'll have to go their repo and search/look for their joystick code.

Alternatively or in addition you could also take a look at @dpculp's QtJSBSim project to see how he uses Qt etc. to read joystick data and feed it into JSBSim, see - https://github.com/dpculp/qtjsbsim

**Reply by bcoconni** [reply]
> Can you please send me the link to that code in which I can use flight gear.

FlightGear use their own copy of the (very) old library PLIB ([link](https://sourceforge.net/p/flightgear/flightgear/ci/next/tree/3rdparty/joystick/)) to interface with joysticks.
